File: MathsThing_Plugins/src/uk/ac/uea/mathsthing/plugins/functions/Round.java
package uk.ac.uea.mathsthing.plugins.functions;

import java.math.BigDecimal;

import uk.ac.uea.mathsthing.IPlugin.IFunctionPlugin;
import uk.ac.uea.mathsthing.util.FormulaException;

public class Round extends IFunctionPlugin {
	
	@Override
	public String getName()
	{
		return "round";
	}

	@Override
	public BigDecimal function(BigDecimal input) throws FormulaException
	{
		return new BigDecimal(Math.round(input.doubleValue()));
	}

}

File: MathsThing_Plugins/src/uk/ac/uea/mathsthing/plugins/functions/Round.scala
package uk.ac.uea.mathsthing.plugins.functions

import uk.ac.uea.mathsthing.IPlugin.IFunctionPlugin
import java.math.BigDecimal
import uk.ac.uea.mathsthing.Functions

class Round extends IFunctionPlugin {

  def getName : String = "round"
    
  def function(input: BigDecimal) : BigDecimal = {
    val result:BigDecimal = new BigDecimal(Math.round(input.doubleValue))
    result
  }
}